diff --git a/init-only-entrypoint.sh b/init-only-entrypoint.sh
index 98a6e66..8997126 100755
--- a/init-only-entrypoint.sh
+++ b/init-only-entrypoint.sh
@@ -77,7 +77,7 @@ if [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
 	# Get config
 	DATADIR="$(_get_config 'datadir' "$@")"
 
-	if [ ! -d "$DATADIR/mysql" ]; then
+	if [ ! -d "$DATADIR/mysql/mysql" ]; then
 		file_env 'MYSQL_ROOT_PASSWORD'
 		if [ -z "$MYSQL_ROOT_PASSWORD" -a -z "$MYSQL_ALLOW_EMPTY_PASSWORD" -a -z "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then
 			echo >&2 'error: database is uninitialized and password option is not specified '
@@ -187,4 +187,6 @@ if [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
 	fi
 fi
 
-exec "$@"
+#exec "$@"
+# Modified to support datadir building
+echo 'Init completed, exiting'
